netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -360,15 +358,24 @@ private int getAndIncrementNextStreamId() {
     return id;
   }
 
+  private Http2Stream requireHttp2Stream(int streamId) throws Http2Exception {
+    Http2Stream stream = connection().stream(streamId);
+    if (stream == null) {
```

Comment:
If it should be impossible except for bug in our code, an AssertionError would more effectively communicate that. Right now I can't tell if we are avoiding buggy remotes.